### ISSUE TITLE
testConrec.cpp:130 fails on ARM64

### ISSUE THREAD

**Issue by bjonnh-work** [rdkit/rdkit#7083]
**Describe the bug**
While building ARM versions of RDKIT on our CI, we see that testConrec.cpp fails
with:
  CHECK( contours.size() == 74 )
with expansion:
  179 == 74

**To Reproduce**
Build on Linux ARM64 and run ctests.
We use those parameters for cmake:
```
      -DCMAKE_BUILD_TYPE=Release \
      -DRDK_BUILD_CAIRO_SUPPORT=ON \
      -DRDK_BUILD_INCHI_SUPPORT=ON \
      -DRDK_BUILD_PGSQL=$1 \  # This doesn't change anything on or off
      -DRDK_BUILD_PYTHON_WRAPPERS=$2 \ # This doesn't change anything on or off
      -DRDK_INSTALL_INTREE=ON \
      -DRDK_INSTALL_STATIC_LIBS=OFF \
      -DRDK_BUILD_CPP_TESTS=ON \
      -DRDK_BUILD_THREADSAFE_SSS=ON \
      -DRDK_TEST_MULTITHREADED=ON \
      -DRDK_BUILD_CFFI_LIB=ON \
      -DBoost_NO_SYSTEM_PATHS=ON \
      -DBoost_NO_BOOST_CMAKE=TRUE \
      -DRDK_BOOST_PYTHON3_NAME=python311 \
      -DPYTHON_EXECUTABLE=${CONDA_PREFIX}/bin/python3 \
      -DCMAKE_INCLUDE_PATH="${CONDA_PREFIX}/include" \
      -DCMAKE_LIBRARY_PATH="${CONDA_PREFIX}/lib" \
      -DCMAKE_POLICY_DEFAULT_CMP0074=NEW ${EXTRA_FLAGS}
```

**Expected behavior**

Getting the same float behavior on ARM64 and AMD64 is sometimes tricky. We would not expect such a test to fail. We have had a few tests to fix in the past for similar reasons. 
Similar issues have been reported:
https://github.com/golang/go/issues/36536
https://eclecticlight.co/2021/07/19/are-there-flaws-in-some-arm64-instructions/

**Configuration (please complete the following information):**
 - RDKit version: master 
 - OS: Ubuntu 22.04
 - Python version (if relevant): Conda's Python 3.11
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? Compiled ourselves

**Comment by greglandrum** [comment]
I've actually also seen this on an Intel Xeon-based system too.
I can take a look and see if I can track it down (I guess it's rounding error in the algorithm)

**Comment by greglandrum** [comment]
That was easy... it was just a zero-tolerance problem in the test code itself. Should be fixed in #7084

**Comment by bjonnh-work** [comment]
Thanks a lot, that was a quick fix!
